features/fout-meldingen.feature

```diff
@@ -0,0 +1,168 @@
+#language: nl
+Functionaliteit: Gebruiksvriendelijke foutmeldingen
+
+    Als developer van een consumer applicatie
+    Wil ik gebruiksvriendelijke en duidelijke foutmeldingen
+    Zodat ik deze één op één kan gebruiken in de UI van mijn applicatie
+
+Rule: minimaal één optionele parameter is opgegeven
+
+    Scenario: Er zijn geen parameters opgegeven
+        Als '/wozobjecten' wordt aangeroepen
+        Dan bevat de response de volgende kenmerken
+        | naam   | waarde                                                                                                                                                                         |
+        | title  | Tenminste één parameter moet worden opgegeven.                                                                                                                                 |
+        | status | 400                                                                                                                                                                            |
+        | detail | Er moet minimaal één van de parameters 'rsin', 'kvkNummer', 'adresseerbaarObjectIdentificatie', 'nummeraanduidingIdentificatie' of 'postcode' met 'huisnummer' worden opgegeven |
+        En bevat de response geen invalidParams
+
+Rule: alleen gespecificeerde parameters mogen worden opgegeven
+
+    Abstract Scenario: Niet gespecificeerd parameter
+        Als '/wozobjecten?<parameter>=123456789' wordt aangeroepen
+        Dan bevat de response de volgende kenmerken
+        | naam   | waarde                                                     |
+        | title  | Een of meerdere parameters zijn niet correct.              |
+        | status | 400                                                        |
+        | detail | Zoeken met parameter '<parameter>' wordt niet ondersteund. |
+        En bevat de response de volgende invalidParams
+        | name        | reason                 |
+        | <parameter> | wordt niet ondersteund |
+
+        Voorbeelden:
+        | description                                                                      | parameter     |
+        | gebruik van een WOZ-object kenmerk die niet wordt ondersteund als zoek parameter | identificatie |
+        | willekeurig string als zoek parameter                                            | bestaatniet   |
+
+Rule: opgegeven parameter(s) heeft een waarde
+
+    Scenario: Er is geen waarde voor één parameter opgegeven
+        Als '/wozobjecten?rsin=' wordt aangeroepen
```

Comment:
hadden we dan in de specificaties niet bij json een minLength (van bijv. 1) moeten definiëren? En dus als toelichting "Waarde is korter dan minimale lengte 1"
Anders zouden we volgens specificatie zoeken met een leegwaarde moeten ondersteunen.
zoeken met /wozobjecten?rsin= moet dan alle wozobjecten leveren met een lege rsin

Reply:
Dat is wel restful. Vraag is wat de functionele wens is. Lijkt mij geen zinnige zoekactie, dus lijkt mij jouw voorstel van een minLength van 1 uitstekend. @CathyDingemanse is zoeken op een lege rsin gewenste functionaliteit ?

Reply:
> Dat is wel restful. Vraag is wat de functionele wens is. Lijkt mij geen zinnige zoekactie, dus lijkt mij jouw voorstel van een minLength van 1 uitstekend. @CathyDingemanse is zoeken op een lege rsin gewenste functionaliteit ?


Nee. Melvin geeft aan dat er een foutmelding volgt.

Reply:
Maar een identificatie van 1 cijfer of 1 letter is toch ook geen geldige identificatie?

Reply:
Dit wordt nu een heel andere discussie. Het scenario is "Er is geen waarde voor één parameter opgegeven.". 
We zouden aan het RSIN natuurlijk ook een pattern kunnen toevoegen en dan volgt er een foutmelding conform de beschrijving in common.